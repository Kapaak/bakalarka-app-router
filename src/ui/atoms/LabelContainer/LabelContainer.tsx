import { cva, VariantProps } from "class-variance-authority";
const labelContainerVariant = cva(["flex"], {
  variants: {
    direction: {
      horizontal: "",
      vertical: "flex-col",
    },
  },
  defaultVariants: {
    direction: "vertical",
  },
});

export type LabelContainerProps = React.ButtonHTMLAttributes<HTMLDivElement> &
  VariantProps<typeof labelContainerVariant> & {
    label: string;
  };

export const LabelContainer = ({
  direction,
  label,
  children,
  className,
}: LabelContainerProps) => {
  return (
    <div className={labelContainerVariant({ direction, className })}>
      <label>{label}</label>
      {children}
    </div>
  );
};
