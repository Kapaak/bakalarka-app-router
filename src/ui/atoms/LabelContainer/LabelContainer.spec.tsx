import { render } from "@testing-library/react";

import { LabelContainer } from "./LabelContainer";

describe("LabelContainer", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<LabelContainer label="" />);

    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
