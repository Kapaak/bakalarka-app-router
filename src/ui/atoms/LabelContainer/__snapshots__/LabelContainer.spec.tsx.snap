// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelContainer should render successfully 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <label />
    </div>
  </div>
</body>
`;
